diff --git a/15.10.2020.zip b/15.10.2020.zip
new file mode 100644
index 0000000..9ee8920
Binary files /dev/null and b/15.10.2020.zip differ
diff --git a/App.js b/App.js
index 62a2cf6..e14429e 100644
--- a/App.js
+++ b/App.js
@@ -138,25 +138,11 @@ export default class App extends React.Component {
           }, 30000);
         } else {
           store.loading = false;
+          console.warn('No')
           store.removeJWT();
-          // this.navigator &&
-          //   this.navigator.dispatch(
-          //     StackActions.reset({
-          //       index: 0,
-          //       actions: [NavigationActions.navigate({ routeName: 'SignUp' })],
-          //     }),
-          //   );
         }
       } else {
         store.loading = false;
-
-        // this.navigator &&
-        //   this.navigator.dispatch(
-        //     StackActions.reset({
-        //       index: 0,
-        //       actions: [NavigationActions.navigate({ routeName: 'SignUp' })],
-        //     }),
-        //   );
       }
     }
   }
@@ -173,22 +159,25 @@ export default class App extends React.Component {
   }

   render() {
-    if(store.agreement) {
-      this.navigator &&
-        this.navigator.dispatch(
-          StackActions.reset({
-            index: 0,
-            actions: [NavigationActions.navigate({ routeName: 'SignIn' })],
-          }),
-        );
-    }
+    // if(store.agreement) {
+    //   this.navigator &&
+    //     this.navigator.dispatch(
+    //       StackActions.reset({
+    //         index: 0,
+    //         actions: [NavigationActions.navigate({ routeName: 'SignIn' })],
+    //       }),
+    //     );
+    // }
     if (!this.state.isReady) {
-      return <AppLoading startAsync={this._loadAssetsAsync.bind(this)} onFinish={() => this.setState({ isReady: true })} onError={console.warn} />;
+      return <AppLoading startAsync={this._loadAssetsAsync.bind(this)}
+       onFinish={() => this.setState({ isReady: true })}
+        onError={console.warn} />;
     }

     return (
       <View style={{ flex: 1 }}>
         <Spinner key="spinner" visible={store.loading} />
+        {/* <Spin store={store} key='spinner'/> */}
         <AppContainer
           key="navigation"
           screenProps={store}
@@ -200,3 +189,11 @@ export default class App extends React.Component {
     );
   }
 }
+
+@observer
+class Spin extends React.Component
+{
+  render() {
+    return <Spinner visible={this.props.store.visible} />
+  }
+}
diff --git a/debug.log b/debug.log
new file mode 100644
index 0000000..93cb47d
--- /dev/null
+++ b/debug.log
@@ -0,0 +1 @@
+[1015/205612.605:ERROR:directory_reader_win.cc(43)] FindFirstFile: ╤шёЄхьх эх єфрхЄё  эрщЄш єърчрээ√щ яєЄ№. (0x3)
diff --git a/screens/Avatar.js b/screens/Avatar.js
index 2d1df78..55df664 100644
--- a/screens/Avatar.js
+++ b/screens/Avatar.js
@@ -11,9 +11,13 @@ export default class Avatar extends React.Component {
   }

   _pickImage = async () => {
-    store.loading = true
+    // store.loading = true
     try {
       let localUri = await store.takePhotoAndUpload();
+      // if(localUri) {
+        console.log(localUri)
+
+      // }
       let filename = localUri.split('/').pop();
       let match = /\.(\w+)$/.exec(filename);
       let type = match ? `image/${match[1]}` : `image`;
@@ -21,28 +25,35 @@ export default class Avatar extends React.Component {
       let formData = new FormData();
       formData.append('action', 'upload-file');
       formData.append('file', { uri: localUri, name: filename, type });
-
-      const response = await fetch(store.apiUrl, {
+      console.log(formData)
+      console.log("before fetch", type)
+      let response = await fetch(store.apiUrl, {
         method: 'POST',
         body: formData,
         header: {
           'content-type': 'multipart/form-data',
         },
-      });
+      })
+      // .then(result => console.warn(result))
+      // .catch(e => console.warn(e))
+      console.warn("response 1", response.ok == true)
       if (response.ok) {
         const fileReq = await response.json();
+        console.warn(fileReq)
         if (fileReq.success && fileReq.data.src) {
           let attachmentSrc = fileReq.data.src;
           let result = await store.apiCall('change-customer', {
             id: store.userId,
             avatar: attachmentSrc,
           });
+          console.warn("response", result)
           if (result) {
-            this.store.avatar = localUri;
+            setTimeout(() => store.avatar = localUri, 1500)
+            // p.navigation.navigate("Avatar", )
           }
         }
       }
-      this.store.loading = false;
+      // this.store.loading = false;
     } catch (e) {
       console.log(e)
       // this.store.loading = false;
diff --git a/screens/SignIn.js b/screens/SignIn.js
index 137d536..f206af3 100644
--- a/screens/SignIn.js
+++ b/screens/SignIn.js
@@ -23,7 +23,7 @@ export default class SignIn extends React.Component {
         super(props)
         this.state = {
             hidePassword: true,
-            currentField: 'signInEmail',
+            // currentField: 'signInEmail',
             email: '',
             password: '',
         }
@@ -43,25 +43,28 @@ export default class SignIn extends React.Component {
                     <View style={{flex: 1, marginRight:p.navigation.isFirstRouteInParent() ? 0:36}}></View>
                 </View>
                 <Toast  ref="toast"
-                        position='top'/>
+                        position='top'
+                        userNativeDriver={true}/>
                 <ScrollView
                     style={{ width: '100%' }}
                     contentContainerStyle={styles.container}
                     keyboardShouldPersistTaps={'handled'}
                 >
                     <View style={[styles.formAuth,
-                        this.state.currentField == "signInEmail"
-                            ? styles.formAuthActive : null]}>
+                        // this.state.currentField == "signInEmail"
+                        //     ? styles.formAuthActive : null
+                            ]}>
+
                         <TextInput
                             style={styles.formAuthValue}
                             keyboardType={'email-address'}
                             placeholder={store.messages.email}
-                            onFocus={(e) => {
-                                this.setState({currentField: "signInEmail"})
-                            }}
-                            onBlur={(e) => {
-                                this.setState({currentField: null})
-                            }}
+                            // onFocus={(e) => {
+                            //     this.setState({currentField: "signInEmail"})
+                            // }}
+                            // onBlur={(e) => {
+                            //     this.setState({currentField: null})
+                            // }}
                             autoFocus = {true}
                             onChangeText={email =>{
                                 this.setState({email})
@@ -70,20 +73,21 @@ export default class SignIn extends React.Component {
                         />
                     </View>
                     <View style={[styles.formAuth,
-                        this.state.currentField == "signInPassword"
-                        ? styles.formAuthActive : null]}>
+                        // this.state.currentField == "signInPassword"
+                        // ? styles.formAuthActive : null
+                        ]}>
                         <View style={{paddingHorizontal:25,
                             flexDirection:'row'}}>
                             <TextInput
                                 style={styles.formAuthValue}
                                 placeholder={store.messages.password}

-                                onFocus={(e) => {
-                                    this.setState({currentField: "signInPassword"})
-                                }}
-                                onBlur={(e) => {
-                                    this.setState({currentField: null})
-                                }}
+                                // onFocus={(e) => {
+                                //     this.setState({currentField: "signInPassword"})
+                                // }}
+                                // onBlur={(e) => {
+                                //     this.setState({currentField: null})
+                                // }}
                                 onChangeText={password => {
                                     this.setState({password});
                                 }}
@@ -109,10 +113,16 @@ export default class SignIn extends React.Component {
                         onPress={async () => {
                             if (validateEmail(this.state.email) && this.state.password.length > 5) {
                                 try {
+                                    store.loading = true
                                     let result = await store.apiCall('auth-customer',{
                                         email: this.state.email,
                                         password: this.state.password
                                     });
+                                    // setTimeout(() => {
+                                    //     console.log(store.loading)
+                                    //     store.loading = false
+                                    // }, 3000)
+                                    store.loading = false
                                     if (!result.success) {
                                         this.refs['toast'].show(result.errorMessage, 2000);
                                     } else {
diff --git a/screens/SignUp.js b/screens/SignUp.js
index 1a14ab9..dd17306 100644
--- a/screens/SignUp.js
+++ b/screens/SignUp.js
@@ -160,7 +160,7 @@ function validateEmail(email) {
           </TouchableOpacity>
           <KeyboardSpacer/>
         </ScrollView>
-        <Toast ref="toast" position='top'/>
+        <Toast ref="toast" position='top' userNativeDriver={true}/>
       </View>
     )
   }
diff --git a/screens/Splash.js b/screens/Splash.js
index 3e9fe52..5b66da6 100644
--- a/screens/Splash.js
+++ b/screens/Splash.js
@@ -1,6 +1,7 @@
 import React from 'react';
 import styles, { colors } from '../styles/index';
 import { Text, View, Image, ScrollView, TouchableOpacity } from 'react-native';
+import { StackActions, NavigationActions } from 'react-navigation';
 import { observer } from 'mobx-react';

 @observer
@@ -27,91 +28,105 @@ export default class Splash extends React.Component {
           </Text>
         </View>
         {!store.jwt
-          ? <View style={styles.splashUserAgreement}>
-              <ScrollView style={{ marginTop: 5}}>
-                <View>
-                  <Text style={[styles.splashText, styles.splashUserAgreementText]}>
-                    <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М╤Б╨║╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡</Text>{"\n\n"}
-                    ╨Э╨░╤Б╤В╨╛╤П╤Й╨╡╨╡ ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М╤Б╨║╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ (╨┤╨░╨╗╨╡╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡) ╤А╨╡╨│╤Г╨╗╨╕╤А╤Г╨╡╤В ╨╛╤В╨╜╨╛╤И╨╡╨╜╨╕╤П ╨╝╨╡╨╢╨┤╤Г iChazy (╨╕╨╗╨╕
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П) ╤Б ╨╛╨┤╨╜╨╛╨╣ ╤Б╤В╨╛╤А╨╛╨╜╤Л ╨╕ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╝ ╨╝╨╛╨▒╨╕╨╗╤М╨╜╨╛╨│╨╛ ╨┐╤А╨╕╨╗╨╛╨╢╨╡╨╜╨╕╤П (╨┤╨░╨╗╨╡╨╡ ╨Ь╨Я) ╤Б ╨┤╤А╤Г╨│╨╛╨╣.{"\n"}
-                    ╨Ь╨Я iChazy ╨╜╨╡ ╤П╨▓╨╗╤П╨╡╤В╤Б╤П ╤Б╤А╨╡╨┤╤Б╤В╨▓╨╛╨╝ ╨╝╨░╤Б╤Б╨╛╨▓╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕.{"\n"}
-                    {"\n"}
-                    ╨Ш╤Б╨┐╨╛╨╗╤М╨╖╤Г╤П ╨Ь╨Я, ╨Т╤Л ╤Б╨╛╨│╨╗╨░╤И╨░╨╡╤В╨╡╤Б╤М ╤Б ╤Г╤Б╨╗╨╛╨▓╨╕╤П╨╝╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П.{"\n"}
-                    ╨Х╤Б╨╗╨╕ ╨Т╤Л ╨╜╨╡ ╤Б╨╛╨│╨╗╨░╤Б╨╜╤Л ╤Б ╤Г╤Б╨╗╨╛╨▓╨╕╤П╨╝╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П, ╨╜╨╡ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╤Г╨╣╤В╨╡ ╨Ь╨Я iChazy!{"\n"}
-                    {"\n"}{"\n"}
-                    <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╤А╨╡╨┤╨╝╨╡╤В ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П</Text>{"\n\n"}
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨┐╤А╨╡╨┤╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╤В ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤О ╨┐╤А╨░╨▓╨╛ ╨╜╨░ ╤А╨░╨╖╨╝╨╡╤Й╨╡╨╜╨╕╨╡ ╨▓ ╨Ь╨Я ╤Б╨╗╨╡╨┤╤Г╤О╤Й╨╡╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕:{"\n"}
-                    - ╨в╨╡╨║╤Б╤В╨╛╨▓╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕{"\n"}
-                    - ╨Р╤Г╨┤╨╕╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
-                    - ╨Т╨╕╨┤╨╡╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
-                    - ╨д╨╛╤В╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
-                    {"\n"}
-                    <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╤А╨░╨▓╨░ ╨╕ ╨╛╨▒╤П╨╖╨░╨╜╨╜╨╛╤Б╤В╨╕ ╤Б╤В╨╛╤А╨╛╨╜</Text>{"\n\n"}
-                    ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╕╨╝╨╡╨╡╤В ╨┐╤А╨░╨▓╨╛:{"\n"}
-                    - ╨╛╤Б╤Г╤Й╨╡╤Б╤В╨▓╨╗╤П╤В╤М ╨┐╨╛╨╕╤Б╨║ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨▓ ╨Ь╨Я;{"\n"}
-                    - ╨┐╨╛╨╗╤Г╤З╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨▓ ╨Ь╨Я;{"\n"}
-                    - ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨┤╨╗╤П ╨Ь╨Я;{"\n"}
-                    - ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨╖╨░ ╨┐╤А╨╡╨┤╨╡╨╗╨░╨╝╨╕ ╨Ь╨Я;{"\n"}
-                    - ╨┐╨╛╨┤╨░╨▓╨░╤В╤М ╨╢╨░╨╗╨╛╨▒╤Л ╨╜╨░ ╨╝╨░╤В╨╡╤А╨╕╨░╨╗, ╨║╨╛╤В╨╛╤А╤Л╨╣ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╤Б╤З╨╕╤В╨░╨╡╤В ╨╜╨╡╨┐╤А╨░╨▓╨╛╨╝╨╡╤А╨╜╤Л╨╝ ╨╕╨╗╨╕ ╨╛╤Б╨║╨╛╤А╨▒╨╕╤В╨╡╨╗╤М╨╜╤Л╨╝;{"\n"}
-                    {"\n"}
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╕╨╝╨╡╨╡╤В ╨┐╤А╨░╨▓╨╛:{"\n"}
-                    - ╨┐╨╛ ╤Б╨▓╨╛╨╡╨╝╤Г ╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨╕╤О ╨╕ ╨╜╨╡╨╛╨▒╤Е╨╛╨┤╨╕╨╝╨╛╤Б╤В╨╕ ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М, ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М, ╨╛╤В╨╝╨╡╨╜╤П╤В╤М ╨┐╤А╨░╨▓╨╕╨╗╨░;{"\n"}
-                    - ╨╛╨│╤А╨░╨╜╨╕╤З╨╕╨▓╨░╤В╤М ╨┤╨╛╤Б╤В╤Г╨┐ ╨║ ╨╗╤О╨▒╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨▓ ╨Ь╨Я;{"\n"}
-                    - ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М, ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М, ╤Г╨┤╨░╨╗╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О;{"\n"}
-                    - ╤Г╨┤╨░╨╗╤П╤В╤М ╤Г╤З╨╡╤В╨╜╤Л╨╡ ╨╖╨░╨┐╨╕╤Б╨╕;{"\n"}
-                    - ╨╛╤В╨║╨░╨╖╤Л╨▓╨░╤В╤М ╨▓ ╤А╨╡╨│╨╕╤Б╤В╤А╨░╤Ж╨╕╨╕ ╨▒╨╡╨╖ ╨╛╨▒╤К╤П╤Б╨╜╨╡╨╜╨╕╤П ╨┐╤А╨╕╤З╨╕╨╜;{"\n"}
-                    {"\n"}
-                    ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╛╨▒╤П╨╖╤Г╨╡╤В╤Б╤П:{"\n"}
-                    - ╨╛╨▒╨╡╤Б╨┐╨╡╤З╨╕╨▓╨░╤В╤М ╤Б╨╛╤Е╤А╨░╨╜╨╜╨╛╤Б╤В╤М ╨╗╨╕╤З╨╜╤Л╤Е ╨┤╨░╨╜╨╜╤Л╤Е ╨╛╤В ╨┤╨╛╤Б╤В╤Г╨┐╨░ ╤В╤А╨╡╤В╤М╨╕╤Е ╨╗╨╕╤Ж;{"\n"}
-                    - ╨╜╨╡ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╨║╨╛╤В╨╛╤А╨░╤П ╨╜╨░╨┐╤А╨░╨▓╨╗╨╡╨╜╨░ ╨╜╨░ ╨┐╤А╨╛╨┐╨░╨│╨░╨╜╨┤╤Г ╨▓╨╛╨╣╨╜╤Л, ╤А╨░╨╖╨╢╨╕╨│╨░╨╜╨╕╨╡ ╨╜╨░╤Ж╨╕╨╛╨╜╨░╨╗╤М╨╜╨╛╨╣,
-                    ╤А╨░╤Б╨╛╨▓╨╛╨╣
-                    ╨╕╨╗╨╕ ╤А╨╡╨╗╨╕╨│╨╕╨╛╨╖╨╜╨╛╨╣ ╨╜╨╡╨╜╨░╨▓╨╕╤Б╤В╨╕ ╨╕ ╨▓╤А╨░╨╢╨┤╤Л, ╨░ ╤В╨░╨║╨╢╨╡ ╨╕╨╜╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕, ╨╖╨░ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨┐╤А╨╡╨┤╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨░
-                    ╤Г╨│╨╛╨╗╨╛╨▓╨╜╨░╤П ╨╕╨╗╨╕ ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤В╨╕╨▓╨╜╨░╤П ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М;{"\n"}
-                    - ╨╜╨╡ ╨╜╨░╤А╤Г╤И╨░╤В╤М ╤А╨░╨▒╨╛╤В╨╛╤Б╨┐╨╛╤Б╨╛╨▒╨╜╨╛╤Б╤В╤М ╨Ь╨Я;{"\n"}
-                    - ╨╜╨╡ ╤Б╨╛╨▓╨╡╤А╤И╨░╤В╤М ╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П, ╨╜╨░╨┐╤А╨░╨▓╨╗╨╡╨╜╨╜╤Л╨╡ ╨╜╨░ ╨▓╨▓╨╡╨┤╨╡╨╜╨╕╨╡ ╨┤╤А╤Г╨│╨╕╤Е ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╣ ╨▓ ╨╖╨░╨▒╨╗╤Г╨╢╨┤╨╡╨╜╨╕╨╡;{"\n"}
-                    - ╨╜╨╡ ╤А╨░╨╖╨╝╨╡╤Й╨░╤В╤М ╨╝╨░╤В╨╡╤А╨╕╨░╨╗╤Л ╤А╨╡╨║╨╗╨░╨╝╨╜╨╛╨│╨╛, ╤Н╤А╨╛╤В╨╕╤З╨╡╤Б╨║╨╛╨│╨╛, ╨┐╨╛╤А╨╜╨╛╨│╤А╨░╤Д╨╕╤З╨╡╤Б╨║╨╛╨│╨╛ ╨╕╨╗╨╕ ╨╛╤Б╨║╨╛╤А╨▒╨╕╤В╨╡╨╗╤М╨╜╨╛╨│╨╛ ╤Е╨░╤А╨░╨║╤В╨╡╤А╨░, ╨░
-                    ╤В╨░╨║╨╢╨╡ ╨╕╨╜╤Г╤О ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╤А╨░╨╖╨╝╨╡╤Й╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨╖╨░╨┐╤А╨╡╤Й╨╡╨╜╨╛ ╨╕╨╗╨╕ ╨┐╤А╨╛╤В╨╕╨▓╨╛╤А╨╡╤З╨╕╤В ╨╜╨╛╤А╨╝╨░╨╝ ╨┤╨╡╨╣╤Б╤В╨▓╤Г╤О╤Й╨╡╨│╨╛
-                    ╨╖╨░╨║╨╛╨╜╨╛╨┤╨░╤В╨╡╨╗╤М╤Б╤В╨▓╨░ ╨а╨д;{"\n"}
-                    - ╨╜╨╡ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╤М ╤Б╨║╤А╨╕╨┐╤В╤Л (╨┐╤А╨╛╨│╤А╨░╨╝╨╝╤Л) ╨┤╨╗╤П ╨░╨▓╤В╨╛╨╝╨░╤В╨╕╨╖╨╕╤А╨╛╨▓╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨▒╨╛╤А╨░ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨╕/╨╕╨╗╨╕ ╨▓╨╖╨░╨╕╨╝╨╛╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П ╤Б
-                    ╨Ь╨Я;{"\n"}
-                    {"\n"}
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╛╨▒╤П╨╖╤Г╨╡╤В╤Б╤П:{"\n"}
-                    - ╨┐╨╛╨┤╨┤╨╡╤А╨╢╨╕╨▓╨░╤В╤М ╤А╨░╨▒╨╛╤В╨╛╤Б╨┐╨╛╤Б╨╛╨▒╨╜╨╛╤Б╤В╤М ╨Ь╨Я ╨╖╨░ ╨╕╤Б╨║╨╗╤О╤З╨╡╨╜╨╕╨╡╨╝ ╤Б╨╗╤Г╤З╨░╨╡╨▓, ╨║╨╛╨│╨┤╨░ ╤Н╤В╨╛ ╨╜╨╡╨▓╨╛╨╖╨╝╨╛╨╢╨╜╨╛ ╨┐╨╛ ╨╜╨╡╨╖╨░╨▓╨╕╤Б╤П╤Й╨╕╨╝ ╨╛╤В
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╨╕ ╨┐╤А╨╕╤З╨╕╨╜╨░╨╝;{"\n"}
-                    - ╨╛╤Б╤Г╤Й╨╡╤Б╤В╨▓╨╗╤П╤В╤М ╤А╨░╨╖╨╜╨╛╤Б╤В╨╛╤А╨╛╨╜╨╜╤О╤О ╨╖╨░╤Й╨╕╤В╤Г ╤Г╤З╨╡╤В╨╜╨╛╨╣ ╨╖╨░╨┐╨╕╤Б╨╕ ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤П;{"\n"}
-                    - ╨╖╨░╤Й╨╕╤Й╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨╛╨│╤А╨░╨╜╨╕╤З╨╡╨╜╨╛ ╨╕╨╗╨╕ ╨╖╨░╨┐╤А╨╡╤Й╨╡╨╜╨╛ ╨╖╨░╨║╨╛╨╜╨░╨╝╨╕ ╨┐╤Г╤В╨╡╨╝ ╨▓╤Л╨╜╨╡╤Б╨╡╨╜╨╕╤П
-                    ╨┐╤А╨╡╨┤╤Г╨┐╤А╨╡╨╢╨┤╨╡╨╜╨╕╤П ╨╗╨╕╨▒╨╛ ╤Г╨┤╨░╨╗╨╡╨╜╨╕╨╡╨╝ ╤Г╤З╨╡╤В╨╜╨╛╨╣ ╨╖╨░╨┐╨╕╤Б╨╕ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤П, ╨╜╨░╤А╤Г╤И╨╕╨▓╤И╨╡╨│╨╛ ╨┐╤А╨░╨▓╨╕╨╗╨░;{"\n"}
-                    {"\n"}
-                    ╨Ю╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╤Б╤В╨╛╤А╨╛╨╜{"\n"}
-                    - ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╗╨╕╤З╨╜╨╛ ╨╜╨╡╤Б╨╡╤В ╨┐╨╛╨╗╨╜╤Г╤О ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╨╖╨░ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╨╡╨╝╤Г╤О ╨╕╨╝ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О;{"\n"}
-                    - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╨╡╤В ╨╜╨╕╨║╨░╨║╨╛╨╣ ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╨╕ ╨╖╨░ ╨┤╨╛╤Б╤В╨╛╨▓╨╡╤А╨╜╨╛╤Б╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕, ╤Б╨║╨╛╨┐╨╕╤А╨╛╨▓╨░╨╜╨╜╨╛╨╣ ╨╕╨╖ ╨┤╤А╤Г╨│╨╕╤Е
-                    ╨╕╤Б╤В╨╛╤З╨╜╨╕╨║╨╛╨▓;{"\n"}
-                    - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╤С╤В ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╨╖╨░ ╨╜╨╡╤Б╨╛╨▓╨┐╨░╨┤╨╡╨╜╨╕╨╡ ╨╛╨╢╨╕╨┤╨░╨╡╨╝╤Л╤Е ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╝ ╨╕ ╤А╨╡╨░╨╗╤М╨╜╨╛ ╨┐╨╛╨╗╤Г╤З╨╡╨╜╨╜╤Л╤Е
-                    ╤Г╤Б╨╗╤Г╨│;{"\n"}
-                    - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╨╡╤В ╨╜╨╕╨║╨░╨║╨╛╨╣ ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╨╕ ╨╖╨░ ╤Г╤Б╨╗╤Г╨│╨╕, ╨┐╤А╨╡╨┤╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╨╝╤Л╨╡ ╤В╤А╨╡╤В╤М╨╕╨╝╨╕ ╨╗╨╕╤Ж╨░╨╝╨╕;{"\n"}
-                    {"\n"}
-                    <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨г╤Б╨╗╨╛╨▓╨╕╤П ╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П</Text>{"\n\n"}
-                    ╨Ф╨░╨╜╨╜╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨▓╤Б╤В╤Г╨┐╨░╨╡╤В ╨▓ ╤Б╨╕╨╗╤Г ╨┐╤А╨╕ ╨╗╤О╨▒╨╛╨╝ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╨╜╨╕╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╨Ь╨Я.{"\n"}
-                    ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨┐╨╡╤А╨╡╤Б╤В╨░╨╡╤В ╨┤╨╡╨╣╤Б╤В╨▓╨╛╨▓╨░╤В╤М ╨┐╤А╨╕ ╨┐╨╛╤П╨▓╨╗╨╡╨╜╨╕╨╕ ╨╡╨│╨╛ ╨╜╨╛╨▓╨╛╨╣ ╨▓╨╡╤А╤Б╨╕╨╕.{"\n"}
-                    ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╤В ╨╖╨░ ╤Б╨╛╨▒╨╛╨╣ ╨┐╤А╨░╨▓╨╛ ╨▓ ╨╛╨┤╨╜╨╛╤Б╤В╨╛╤А╨╛╨╜╨╜╨╡╨╝ ╨┐╨╛╤А╤П╨┤╨║╨╡ ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М ╨┤╨░╨╜╨╜╨╛╨╡ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨┐╨╛ ╤Б╨▓╨╛╨╡╨╝╤Г
-                    ╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨╕╤О.{"\n"}
-                    ╨Я╤А╨╕ ╨╕╨╖╨╝╨╡╨╜╨╡╨╜╨╕╨╕ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П, ╨▓ ╨╜╨╡╨║╨╛╤В╨╛╤А╤Л╤Е ╤Б╨╗╤Г╤З╨░╤П╤Е, ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╝╨╛╨╢╨╡╤В ╨╛╨┐╨╛╨▓╨╡╤Б╤В╨╕╤В╤М ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╣ ╤Г╨┤╨╛╨▒╨╜╤Л╨╝
-                    ╨┤╨╗╤П
-                    ╨╜╨╡╨╡ ╤Б╨┐╨╛╤Б╨╛╨▒╨╛╨╝.{"\n"}
-                  </Text>
-                </View>
-              </ScrollView>
-              <TouchableOpacity
-                onPress={() => {store.agreement = true
-                  this.props.navigation.navigate('SignIn')}}
-                style={[styles.button, styles.tutorialButton]}
-              >
-                <Text style={styles.buttonText}>╨Я╤А╨╛╨┤╨╛╨╗╨╢╨╕╤В╤М</Text>
-              </TouchableOpacity>
-            </View>
+          ? <Agreement/>
           : null
         }
+        <TouchableOpacity
+          onPress={() => {
+            this.props.navigation.navigate('SignIn')
+            this.props.navigation.dispatch(
+              StackActions.reset({
+                index: 0,
+                actions: [NavigationActions.navigate({ routeName: 'SignIn' })],
+              }),
+            )
+          }
+          }
+          style={[styles.button, styles.tutorialButton]}
+        >
+          <Text style={styles.buttonText}>╨Я╤А╨╛╨┤╨╛╨╗╨╢╨╕╤В╤М</Text>
+        </TouchableOpacity>
       </View>
     );
   }
 }
+
+// ╨▓╤Л╨╜╨╡╤Б ╨▓ ╨╛╤В╨┤╨╡╨╗╤М╨╜╤Л╨╣ ╨║╨╛╨╝╨┐╨╛╨╜╨╡╨╜╤В, ╤З╤В╨╛╨▒╤Л ╨│╨╗╨░╨╖╨░ ╨╜╨╡ ╨╝╨╛╨╖╨╛╨╗╨╕╨╗╨╛. ╨Т╨╛╨╛╨▒╤Й╨╡ ╨║╨╛╨╜╨╡╤З╨╜╨╛ ╨╜╨╡
+// ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨▓╤Л╨╜╨╡╤Б╤В╨╕ ╨▒╤Л ╨▓ ╨║╨░╨║╨╛╨╣-╨╜╨╕╨▒╤Г╨┤╤М markdown. ╨Э╨╡
+const Agreement = React.memo(() => (
+  <View style={styles.splashUserAgreement}>
+  <ScrollView style={{ marginTop: 5}}>
+    <View>
+      <Text style={[styles.splashText, styles.splashUserAgreementText]}>
+        <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М╤Б╨║╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡</Text>{"\n\n"}
+        ╨Э╨░╤Б╤В╨╛╤П╤Й╨╡╨╡ ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М╤Б╨║╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ (╨┤╨░╨╗╨╡╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡) ╤А╨╡╨│╤Г╨╗╨╕╤А╤Г╨╡╤В ╨╛╤В╨╜╨╛╤И╨╡╨╜╨╕╤П ╨╝╨╡╨╢╨┤╤Г iChazy (╨╕╨╗╨╕
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П) ╤Б ╨╛╨┤╨╜╨╛╨╣ ╤Б╤В╨╛╤А╨╛╨╜╤Л ╨╕ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╝ ╨╝╨╛╨▒╨╕╨╗╤М╨╜╨╛╨│╨╛ ╨┐╤А╨╕╨╗╨╛╨╢╨╡╨╜╨╕╤П (╨┤╨░╨╗╨╡╨╡ ╨Ь╨Я) ╤Б ╨┤╤А╤Г╨│╨╛╨╣.{"\n"}
+        ╨Ь╨Я iChazy ╨╜╨╡ ╤П╨▓╨╗╤П╨╡╤В╤Б╤П ╤Б╤А╨╡╨┤╤Б╤В╨▓╨╛╨╝ ╨╝╨░╤Б╤Б╨╛╨▓╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕.{"\n"}
+        {"\n"}
+        ╨Ш╤Б╨┐╨╛╨╗╤М╨╖╤Г╤П ╨Ь╨Я, ╨Т╤Л ╤Б╨╛╨│╨╗╨░╤И╨░╨╡╤В╨╡╤Б╤М ╤Б ╤Г╤Б╨╗╨╛╨▓╨╕╤П╨╝╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П.{"\n"}
+        ╨Х╤Б╨╗╨╕ ╨Т╤Л ╨╜╨╡ ╤Б╨╛╨│╨╗╨░╤Б╨╜╤Л ╤Б ╤Г╤Б╨╗╨╛╨▓╨╕╤П╨╝╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П, ╨╜╨╡ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╤Г╨╣╤В╨╡ ╨Ь╨Я iChazy!{"\n"}
+        {"\n"}{"\n"}
+        <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╤А╨╡╨┤╨╝╨╡╤В ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П</Text>{"\n\n"}
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨┐╤А╨╡╨┤╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╤В ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤О ╨┐╤А╨░╨▓╨╛ ╨╜╨░ ╤А╨░╨╖╨╝╨╡╤Й╨╡╨╜╨╕╨╡ ╨▓ ╨Ь╨Я ╤Б╨╗╨╡╨┤╤Г╤О╤Й╨╡╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕:{"\n"}
+        - ╨в╨╡╨║╤Б╤В╨╛╨▓╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕{"\n"}
+        - ╨Р╤Г╨┤╨╕╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
+        - ╨Т╨╕╨┤╨╡╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
+        - ╨д╨╛╤В╨╛╨╝╨░╤В╨╡╤А╨╕╨░╨╗╨╛╨▓{"\n"}
+        {"\n"}
+        <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨Я╤А╨░╨▓╨░ ╨╕ ╨╛╨▒╤П╨╖╨░╨╜╨╜╨╛╤Б╤В╨╕ ╤Б╤В╨╛╤А╨╛╨╜</Text>{"\n\n"}
+        ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╕╨╝╨╡╨╡╤В ╨┐╤А╨░╨▓╨╛:{"\n"}
+        - ╨╛╤Б╤Г╤Й╨╡╤Б╤В╨▓╨╗╤П╤В╤М ╨┐╨╛╨╕╤Б╨║ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨▓ ╨Ь╨Я;{"\n"}
+        - ╨┐╨╛╨╗╤Г╤З╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨▓ ╨Ь╨Я;{"\n"}
+        - ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨┤╨╗╤П ╨Ь╨Я;{"\n"}
+        - ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О ╨╖╨░ ╨┐╤А╨╡╨┤╨╡╨╗╨░╨╝╨╕ ╨Ь╨Я;{"\n"}
+        - ╨┐╨╛╨┤╨░╨▓╨░╤В╤М ╨╢╨░╨╗╨╛╨▒╤Л ╨╜╨░ ╨╝╨░╤В╨╡╤А╨╕╨░╨╗, ╨║╨╛╤В╨╛╤А╤Л╨╣ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╤Б╤З╨╕╤В╨░╨╡╤В ╨╜╨╡╨┐╤А╨░╨▓╨╛╨╝╨╡╤А╨╜╤Л╨╝ ╨╕╨╗╨╕ ╨╛╤Б╨║╨╛╤А╨▒╨╕╤В╨╡╨╗╤М╨╜╤Л╨╝;{"\n"}
+        {"\n"}
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╕╨╝╨╡╨╡╤В ╨┐╤А╨░╨▓╨╛:{"\n"}
+        - ╨┐╨╛ ╤Б╨▓╨╛╨╡╨╝╤Г ╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨╕╤О ╨╕ ╨╜╨╡╨╛╨▒╤Е╨╛╨┤╨╕╨╝╨╛╤Б╤В╨╕ ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М, ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М, ╨╛╤В╨╝╨╡╨╜╤П╤В╤М ╨┐╤А╨░╨▓╨╕╨╗╨░;{"\n"}
+        - ╨╛╨│╤А╨░╨╜╨╕╤З╨╕╨▓╨░╤В╤М ╨┤╨╛╤Б╤В╤Г╨┐ ╨║ ╨╗╤О╨▒╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨▓ ╨Ь╨Я;{"\n"}
+        - ╤Б╨╛╨╖╨┤╨░╨▓╨░╤В╤М, ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М, ╤Г╨┤╨░╨╗╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О;{"\n"}
+        - ╤Г╨┤╨░╨╗╤П╤В╤М ╤Г╤З╨╡╤В╨╜╤Л╨╡ ╨╖╨░╨┐╨╕╤Б╨╕;{"\n"}
+        - ╨╛╤В╨║╨░╨╖╤Л╨▓╨░╤В╤М ╨▓ ╤А╨╡╨│╨╕╤Б╤В╤А╨░╤Ж╨╕╨╕ ╨▒╨╡╨╖ ╨╛╨▒╤К╤П╤Б╨╜╨╡╨╜╨╕╤П ╨┐╤А╨╕╤З╨╕╨╜;{"\n"}
+        {"\n"}
+        ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╛╨▒╤П╨╖╤Г╨╡╤В╤Б╤П:{"\n"}
+        - ╨╛╨▒╨╡╤Б╨┐╨╡╤З╨╕╨▓╨░╤В╤М ╤Б╨╛╤Е╤А╨░╨╜╨╜╨╛╤Б╤В╤М ╨╗╨╕╤З╨╜╤Л╤Е ╨┤╨░╨╜╨╜╤Л╤Е ╨╛╤В ╨┤╨╛╤Б╤В╤Г╨┐╨░ ╤В╤А╨╡╤В╤М╨╕╤Е ╨╗╨╕╤Ж;{"\n"}
+        - ╨╜╨╡ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╨║╨╛╤В╨╛╤А╨░╤П ╨╜╨░╨┐╤А╨░╨▓╨╗╨╡╨╜╨░ ╨╜╨░ ╨┐╤А╨╛╨┐╨░╨│╨░╨╜╨┤╤Г ╨▓╨╛╨╣╨╜╤Л, ╤А╨░╨╖╨╢╨╕╨│╨░╨╜╨╕╨╡ ╨╜╨░╤Ж╨╕╨╛╨╜╨░╨╗╤М╨╜╨╛╨╣,
+        ╤А╨░╤Б╨╛╨▓╨╛╨╣
+        ╨╕╨╗╨╕ ╤А╨╡╨╗╨╕╨│╨╕╨╛╨╖╨╜╨╛╨╣ ╨╜╨╡╨╜╨░╨▓╨╕╤Б╤В╨╕ ╨╕ ╨▓╤А╨░╨╢╨┤╤Л, ╨░ ╤В╨░╨║╨╢╨╡ ╨╕╨╜╨╛╨╣ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕, ╨╖╨░ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨┐╤А╨╡╨┤╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨░
+        ╤Г╨│╨╛╨╗╨╛╨▓╨╜╨░╤П ╨╕╨╗╨╕ ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤В╨╕╨▓╨╜╨░╤П ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М;{"\n"}
+        - ╨╜╨╡ ╨╜╨░╤А╤Г╤И╨░╤В╤М ╤А╨░╨▒╨╛╤В╨╛╤Б╨┐╨╛╤Б╨╛╨▒╨╜╨╛╤Б╤В╤М ╨Ь╨Я;{"\n"}
+        - ╨╜╨╡ ╤Б╨╛╨▓╨╡╤А╤И╨░╤В╤М ╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П, ╨╜╨░╨┐╤А╨░╨▓╨╗╨╡╨╜╨╜╤Л╨╡ ╨╜╨░ ╨▓╨▓╨╡╨┤╨╡╨╜╨╕╨╡ ╨┤╤А╤Г╨│╨╕╤Е ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╣ ╨▓ ╨╖╨░╨▒╨╗╤Г╨╢╨┤╨╡╨╜╨╕╨╡;{"\n"}
+        - ╨╜╨╡ ╤А╨░╨╖╨╝╨╡╤Й╨░╤В╤М ╨╝╨░╤В╨╡╤А╨╕╨░╨╗╤Л ╤А╨╡╨║╨╗╨░╨╝╨╜╨╛╨│╨╛, ╤Н╤А╨╛╤В╨╕╤З╨╡╤Б╨║╨╛╨│╨╛, ╨┐╨╛╤А╨╜╨╛╨│╤А╨░╤Д╨╕╤З╨╡╤Б╨║╨╛╨│╨╛ ╨╕╨╗╨╕ ╨╛╤Б╨║╨╛╤А╨▒╨╕╤В╨╡╨╗╤М╨╜╨╛╨│╨╛ ╤Е╨░╤А╨░╨║╤В╨╡╤А╨░, ╨░
+        ╤В╨░╨║╨╢╨╡ ╨╕╨╜╤Г╤О ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╤А╨░╨╖╨╝╨╡╤Й╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨╖╨░╨┐╤А╨╡╤Й╨╡╨╜╨╛ ╨╕╨╗╨╕ ╨┐╤А╨╛╤В╨╕╨▓╨╛╤А╨╡╤З╨╕╤В ╨╜╨╛╤А╨╝╨░╨╝ ╨┤╨╡╨╣╤Б╤В╨▓╤Г╤О╤Й╨╡╨│╨╛
+        ╨╖╨░╨║╨╛╨╜╨╛╨┤╨░╤В╨╡╨╗╤М╤Б╤В╨▓╨░ ╨а╨д;{"\n"}
+        - ╨╜╨╡ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╤М ╤Б╨║╤А╨╕╨┐╤В╤Л (╨┐╤А╨╛╨│╤А╨░╨╝╨╝╤Л) ╨┤╨╗╤П ╨░╨▓╤В╨╛╨╝╨░╤В╨╕╨╖╨╕╤А╨╛╨▓╨░╨╜╨╜╨╛╨│╨╛ ╤Б╨▒╨╛╤А╨░ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕ ╨╕/╨╕╨╗╨╕ ╨▓╨╖╨░╨╕╨╝╨╛╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П ╤Б
+        ╨Ь╨Я;{"\n"}
+        {"\n"}
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╛╨▒╤П╨╖╤Г╨╡╤В╤Б╤П:{"\n"}
+        - ╨┐╨╛╨┤╨┤╨╡╤А╨╢╨╕╨▓╨░╤В╤М ╤А╨░╨▒╨╛╤В╨╛╤Б╨┐╨╛╤Б╨╛╨▒╨╜╨╛╤Б╤В╤М ╨Ь╨Я ╨╖╨░ ╨╕╤Б╨║╨╗╤О╤З╨╡╨╜╨╕╨╡╨╝ ╤Б╨╗╤Г╤З╨░╨╡╨▓, ╨║╨╛╨│╨┤╨░ ╤Н╤В╨╛ ╨╜╨╡╨▓╨╛╨╖╨╝╨╛╨╢╨╜╨╛ ╨┐╨╛ ╨╜╨╡╨╖╨░╨▓╨╕╤Б╤П╤Й╨╕╨╝ ╨╛╤В
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╨╕ ╨┐╤А╨╕╤З╨╕╨╜╨░╨╝;{"\n"}
+        - ╨╛╤Б╤Г╤Й╨╡╤Б╤В╨▓╨╗╤П╤В╤М ╤А╨░╨╖╨╜╨╛╤Б╤В╨╛╤А╨╛╨╜╨╜╤О╤О ╨╖╨░╤Й╨╕╤В╤Г ╤Г╤З╨╡╤В╨╜╨╛╨╣ ╨╖╨░╨┐╨╕╤Б╨╕ ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤П;{"\n"}
+        - ╨╖╨░╤Й╨╕╤Й╨░╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О, ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╨╡╨╜╨╕╨╡ ╨║╨╛╤В╨╛╤А╨╛╨╣ ╨╛╨│╤А╨░╨╜╨╕╤З╨╡╨╜╨╛ ╨╕╨╗╨╕ ╨╖╨░╨┐╤А╨╡╤Й╨╡╨╜╨╛ ╨╖╨░╨║╨╛╨╜╨░╨╝╨╕ ╨┐╤Г╤В╨╡╨╝ ╨▓╤Л╨╜╨╡╤Б╨╡╨╜╨╕╤П
+        ╨┐╤А╨╡╨┤╤Г╨┐╤А╨╡╨╢╨┤╨╡╨╜╨╕╤П ╨╗╨╕╨▒╨╛ ╤Г╨┤╨░╨╗╨╡╨╜╨╕╨╡╨╝ ╤Г╤З╨╡╤В╨╜╨╛╨╣ ╨╖╨░╨┐╨╕╤Б╨╕ ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤П, ╨╜╨░╤А╤Г╤И╨╕╨▓╤И╨╡╨│╨╛ ╨┐╤А╨░╨▓╨╕╨╗╨░;{"\n"}
+        {"\n"}
+        ╨Ю╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╤Б╤В╨╛╤А╨╛╨╜{"\n"}
+        - ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╤М ╨╗╨╕╤З╨╜╨╛ ╨╜╨╡╤Б╨╡╤В ╨┐╨╛╨╗╨╜╤Г╤О ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╨╖╨░ ╤А╨░╤Б╨┐╤А╨╛╤Б╤В╤А╨░╨╜╤П╨╡╨╝╤Г╤О ╨╕╨╝ ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╤О;{"\n"}
+        - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╨╡╤В ╨╜╨╕╨║╨░╨║╨╛╨╣ ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╨╕ ╨╖╨░ ╨┤╨╛╤Б╤В╨╛╨▓╨╡╤А╨╜╨╛╤Б╤В╤М ╨╕╨╜╤Д╨╛╤А╨╝╨░╤Ж╨╕╨╕, ╤Б╨║╨╛╨┐╨╕╤А╨╛╨▓╨░╨╜╨╜╨╛╨╣ ╨╕╨╖ ╨┤╤А╤Г╨│╨╕╤Е
+        ╨╕╤Б╤В╨╛╤З╨╜╨╕╨║╨╛╨▓;{"\n"}
+        - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╤С╤В ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╤М ╨╖╨░ ╨╜╨╡╤Б╨╛╨▓╨┐╨░╨┤╨╡╨╜╨╕╨╡ ╨╛╨╢╨╕╨┤╨░╨╡╨╝╤Л╤Е ╨Я╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╝ ╨╕ ╤А╨╡╨░╨╗╤М╨╜╨╛ ╨┐╨╛╨╗╤Г╤З╨╡╨╜╨╜╤Л╤Е
+        ╤Г╤Б╨╗╤Г╨│;{"\n"}
+        - ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╜╨╡ ╨╜╨╡╤Б╨╡╤В ╨╜╨╕╨║╨░╨║╨╛╨╣ ╨╛╤В╨▓╨╡╤В╤Б╤В╨▓╨╡╨╜╨╜╨╛╤Б╤В╨╕ ╨╖╨░ ╤Г╤Б╨╗╤Г╨│╨╕, ╨┐╤А╨╡╨┤╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╨╝╤Л╨╡ ╤В╤А╨╡╤В╤М╨╕╨╝╨╕ ╨╗╨╕╤Ж╨░╨╝╨╕;{"\n"}
+        {"\n"}
+        <Text style={{fontSize: 18, fontWeight: 'bold'}}>╨г╤Б╨╗╨╛╨▓╨╕╤П ╨┤╨╡╨╣╤Б╤В╨▓╨╕╤П ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П</Text>{"\n\n"}
+        ╨Ф╨░╨╜╨╜╨╛╨╡ ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨▓╤Б╤В╤Г╨┐╨░╨╡╤В ╨▓ ╤Б╨╕╨╗╤Г ╨┐╤А╨╕ ╨╗╤О╨▒╨╛╨╝ ╨╕╤Б╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╨╜╨╕╨╕ ╨┤╨░╨╜╨╜╨╛╨│╨╛ ╨Ь╨Я.{"\n"}
+        ╨б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨┐╨╡╤А╨╡╤Б╤В╨░╨╡╤В ╨┤╨╡╨╣╤Б╤В╨▓╨╛╨▓╨░╤В╤М ╨┐╤А╨╕ ╨┐╨╛╤П╨▓╨╗╨╡╨╜╨╕╨╕ ╨╡╨│╨╛ ╨╜╨╛╨▓╨╛╨╣ ╨▓╨╡╤А╤Б╨╕╨╕.{"\n"}
+        ╨Р╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╛╤Б╤В╨░╨▓╨╗╤П╨╡╤В ╨╖╨░ ╤Б╨╛╨▒╨╛╨╣ ╨┐╤А╨░╨▓╨╛ ╨▓ ╨╛╨┤╨╜╨╛╤Б╤В╨╛╤А╨╛╨╜╨╜╨╡╨╝ ╨┐╨╛╤А╤П╨┤╨║╨╡ ╨╕╨╖╨╝╨╡╨╜╤П╤В╤М ╨┤╨░╨╜╨╜╨╛╨╡ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╨╡ ╨┐╨╛ ╤Б╨▓╨╛╨╡╨╝╤Г
+        ╤Г╤Б╨╝╨╛╤В╤А╨╡╨╜╨╕╤О.{"\n"}
+        ╨Я╤А╨╕ ╨╕╨╖╨╝╨╡╨╜╨╡╨╜╨╕╨╕ ╤Б╨╛╨│╨╗╨░╤И╨╡╨╜╨╕╤П, ╨▓ ╨╜╨╡╨║╨╛╤В╨╛╤А╤Л╤Е ╤Б╨╗╤Г╤З╨░╤П╤Е, ╨░╨┤╨╝╨╕╨╜╨╕╤Б╤В╤А╨░╤Ж╨╕╤П ╨╝╨╛╨╢╨╡╤В ╨╛╨┐╨╛╨▓╨╡╤Б╤В╨╕╤В╤М ╨┐╨╛╨╗╤М╨╖╨╛╨▓╨░╤В╨╡╨╗╨╡╨╣ ╤Г╨┤╨╛╨▒╨╜╤Л╨╝
+        ╨┤╨╗╤П
+        ╨╜╨╡╨╡ ╤Б╨┐╨╛╤Б╨╛╨▒╨╛╨╝.{"\n"}
+      </Text>
+    </View>
+  </ScrollView>
+  </View>
+))
\ No newline at end of file
diff --git a/stores/AppStore.js b/stores/AppStore.js
index 973b8b4..3f32aac 100644
--- a/stores/AppStore.js
+++ b/stores/AppStore.js
@@ -330,7 +330,6 @@ export default class AppStore {
         // ? ImagePicker.MediaTypeOptions.Videos
         // :
         ImagePicker.MediaTypeOptions.Images,
-        // : 'Images',
       allowsEditing: true,
       // quality: type == 'video' ? null : 0.5,
     })
@@ -369,8 +368,9 @@ export default class AppStore {
           this.currentImage = result.uri
         }
         return result.uri
-      } else navigate('Avatar')
-    } else await Permissions.askAsync(Permissions.CAMERA_ROLL)
+      }
+    }
+    return false
   }

   get = function (obj, id) {
